extensions/stackdriver/stackdriver.cc

```diff
@@ -91,6 +99,52 @@ StackdriverOptions StackdriverRootContext::getStackdriverOptions() {
   return options;
 }
 
+FilterHeadersStatus StackdriverContext::onRequestHeaders() {
+  request_info_.start_timestamp = proxy_getCurrentTimeNanoseconds();
+  request_info_.request_protocol = getProtocol(StreamType::Request)->toString();
+  request_info_.destination_service_host =
+      getHeaderMapValue(HeaderMapType::RequestHeaders, kAuthorityHeaderKey)
+          ->toString();
+  request_info_.request_operation =
+      getHeaderMapValue(HeaderMapType::RequestHeaders, kMethodHeaderKey)
+          ->toString();
+  if (getRootContext()->reporterKind() ==
+      PluginConfig::ReporterKind::PluginConfig_ReporterKind_INBOUND) {
+    auto downstream_metadata = getMetadataStruct(
+        Common::Wasm::MetadataType::Request, kDownstreamMetadataKey);
+    ExtractNodeMetadata(downstream_metadata, &request_info_.peer_node_info);
+  }
+  return FilterHeadersStatus::Continue;
+}
+
+FilterDataStatus StackdriverContext::onRequestBody(size_t body_buffer_length,
+                                                   bool) {
+  request_info_.request_size += body_buffer_length;
+  return FilterDataStatus::Continue;
+}
+
+FilterHeadersStatus StackdriverContext::onResponseHeaders() {
+  if (getRootContext()->reporterKind() ==
+      PluginConfig::ReporterKind::PluginConfig_ReporterKind_OUTBOUND) {
+    auto upstream_metadata = getMetadataStruct(
+        Common::Wasm::MetadataType::Request, kUpstreamMetadataKey);
+    ExtractNodeMetadata(upstream_metadata, &request_info_.peer_node_info);
+  }
+  request_info_.end_timestamp = proxy_getCurrentTimeNanoseconds();
+  return FilterHeadersStatus::Continue;
+}
+
+FilterDataStatus StackdriverContext::onResponseBody(size_t body_buffer_length,
+                                                    bool) {
+  request_info_.response_size += body_buffer_length;
```

Comment:
This is already available as `stream_info.bytesReceived()` and `stream_info.bytesSent()`.
This is only OK if is not exposed through API. We need to file an issue to expose it, since it's more efficient to reuse the computation.

Reply:
Added a todo to track this.